subsys/fs/fat_fs.c

```diff
@@ -63,7 +63,35 @@ static int translate_error(int error)
 	return -EIO;
 }
 
-static int fatfs_open(struct fs_file_t *zfp, const char *file_name)
+static int translate_flags(int flags)
+{
+	int fat_mode = 0;
+
+	if (flags & FO_READ) {
+		fat_mode |= FA_READ;
+		flags ^= FO_READ;
+	}
+
+	if (flags & FO_WRITE) {
+		fat_mode |= FA_WRITE;
+		flags ^= FO_WRITE;
+	}
+
+	if (flags & FO_CREATE) {
+		fat_mode |= FA_OPEN_ALWAYS;
+		flags ^= FO_CREATE;
+	}
+
+	if (flags & FO_APPEND) {
+		/* NOTE: Unfortunately FA_OPEN_APPEND includes FA_OPEN_ALWAYS */
+		fat_mode |= FA_OPEN_APPEND;
+		flags ^= FO_APPEND;
+	}
+
```

Comment:
I believe the intention was to warn here about unsupported flags, as in the littlefs version of the same code?
If not, there's no reason to clear bits in `flags`.

Reply:
Will fix.